File: src/main/java/com/example/quizapp/entity/Result.java
package com.example.quizapp.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "results")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Result extends BaseEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false)
	private Long score;

	@Column(nullable = false)
	private Long totalScore;

	@Column(nullable = false)
	private Long totalQuestion;

	@Column(nullable = false)
	private Long timeSpent;

	@Column(nullable = false)
	private Boolean isCompleted;

	@Column(nullable = false)
	private Long correctAnswers;

	@Column(nullable = false)
	private Long incorrectAnswers;

	@Column(nullable = false)
	private Long timesTaken;

	@ManyToOne()
	@JoinColumn(name = "user_id", nullable = false)
	@JsonBackReference
	private User user;

	@ManyToOne()
	@JoinColumn(name = "quiz_id", nullable = false)
	@JsonIgnoreProperties("results")
	private Quiz quiz;
}

File: app/models/result.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel
from uuid import UUID

class Result(BaseModel):
    id: Optional[int] = None
    score: int
    total_score: int
    total_question: int
    time_spent: int
    is_completed: bool
    correct_answers: int
    incorrect_answers: int
    times_taken: int
    user_id: UUID
    quiz_id: int
    is_deleted: bool = False
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
